lib/correios_sigep/logistic_reverse/request_sro.rb

```diff
@@ -15,14 +15,19 @@ def process
       private
       def process_response response
         correios_hash = response.body[:acompanhar_pedido_response][:return]
-        check_sro_errors correios_hash
-        correios_hash[:coleta][:objeto][:numero_etiqueta] || begin raise Models::Errors::SRONotReady; end
+        check_sro_errors(correios_hash)
+        correios_response(correios_hash[:coleta][:objeto])[:numero_etiqueta] || begin raise Models::Errors::SRONotReady; end
+      end
+
+      def correios_response response
+        response.is_a?(Array) ? response.first : response
```

Comment:
I think it makes perfectly sense! I'll open an issue in order not to forget this :smile: